pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -175,17 +176,40 @@ func (c *Controller) Apply(change *sink.Change) error {
 		}
 	}
 
-	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
-	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
 	c.configStoreMu.Unlock()
 
-	if descriptor.Type == model.ServiceEntry.Type {
-		c.serviceEntryEvents(innerStore, prevStore)
+	if descriptor.Type == model.ServiceEntry.Type &&
+		descriptor.Collection == model.ServiceEntry.Collection {
+		for ns, byName := range innerStore {
+			for name, config := range byName {
+				endpoints := []*model.IstioEndpoint{}
+				if se, ok := config.Spec.(*networking.ServiceEntry); ok {
+					for _, ep := range se.Endpoints {
+						for portName, port := range ep.Ports {
+							ep := &model.IstioEndpoint{
+								Address:         ep.Address,
+								EndpointPort:    port,
+								ServicePortName: portName,
+								Labels:          ep.Labels,
+								UID:             fmt.Sprintf("%s.%s", name, ns),
+								// TODO: how to get network and ServiceAccount?
+							}
```

Comment:
service entry endpoints have a network and locality field

Reply:
how about serviceAccount? is that something that needs to be assigned here?

Reply:
Service account is missing in service entry spec. Has to be added

Reply:
Right, missed that. And we had a long debate on the security of trusting whatever the discovery decide to place in the SA - with the conclusion that as long as it's in the same namespace it doesn't really matter since isolation between SA in a namespace is weak anyways. 
I think some past code used a synthetic label. 

Not entirely sure where it should be added - the list of endpoints may get down to zero, and that has nasty effects due to propagation of CDS when an endpoing shows up.  Probably should be added in Service, as a list of all SA that the service should put in CDS, with some code to avoid the flakiness on restarts.